

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Setup &mdash; ncmw  documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Analysis" href="analysis.html" />
    <link rel="prev" title="Overview" href="readme.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> ncmw
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="readme.html">Overview</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Setup</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#Set-default-configurations">Set default configurations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Gapfilling">Gapfilling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#Cleaning-the-model">Cleaning the model</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="analysis.html">Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="community.html">Community</a></li>
<li class="toctree-l1"><a class="reference internal" href="hierarchical_model_composition.html">Hierarchical</a></li>
<li class="toctree-l1"><a class="reference internal" href="script_parameters.html">Script Parameters</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
<li class="toctree-l1"><a class="reference internal" href="changelog.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="api/modules.html">Module Reference</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ncmw</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Setup</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/setup.ipynb.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput,
div.nbinput div.prompt,
div.nbinput div.input_area,
div.nbinput div[class*=highlight],
div.nbinput div[class*=highlight] pre,
div.nboutput,
div.nbinput div.prompt,
div.nbinput div.output_area,
div.nboutput div[class*=highlight],
div.nboutput div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput,
div.nboutput {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput,
    div.nboutput {
        flex-direction: column;
    }
}

/* input container */
div.nbinput {
    padding-top: 5px;
}

/* last container */
div.nblast {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput div.prompt,
div.nboutput div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput div.prompt,
    div.nboutput div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput div.prompt pre,
div.nboutput div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput div.input_area,
div.nboutput div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput div.input_area,
    div.nboutput div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput div.math p {
    text-align: left;
}

/* standard error */
div.nboutput div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast,
.nboutput.nblast {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast + .nbinput {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<div class="section" id="Setup">
<h1>Setup<a class="headerlink" href="#Setup" title="Permalink to this headline">¶</a></h1>
<p>An example of what the model setup can do is shown here. This will be demonstrated on five metabolic models: Moraxella, H. Influenza, S. aureus, S. epidermidis and D. pigrum.</p>
<p>Hence, let’s import some of the methods we will use.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">ncmw.utils</span> <span class="k">import</span> <span class="n">get_models</span>
<span class="kn">from</span> <span class="nn">ncmw.setup_models</span> <span class="k">import</span> <span class="n">set_default_configs_and_snm3_medium</span><span class="p">,</span> <span class="n">create_consistent_model</span><span class="p">,</span> <span class="n">gapfill_medium</span><span class="p">,</span> <span class="n">gapfill_model</span>
<span class="n">models</span> <span class="o">=</span> <span class="n">get_models</span><span class="p">(</span><span class="s2">&quot;models&quot;</span><span class="p">)</span>
<span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">models</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;Growth </span><span class="si">{model.id}</span><span class="s2">: {model.slim_optimize()}&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Growth CarveMe_SNM_gapfilled_model: 0.7022727466098083
Growth himodel: 63.919604406865695
Growth Staphylococcus_epidermidis_ATCC_12228: 0.5746259500792691
Growth DP_83VPs_KB5: 0.28236539253224924
Growth iYS854: 1.3602249869877683
</pre></div></div>
</div>
<p>Notice that the growth values differ strongly. This is because the bounds and the medium differ. So let’s first standardize these!</p>
<div class="section" id="Set-default-configurations">
<h2>Set default configurations<a class="headerlink" href="#Set-default-configurations" title="Permalink to this headline">¶</a></h2>
<p>This package is devoted to the work on nasal microbiomes expressed through the <strong>SNM3</strong> medium. Unfortunately, most metabolic models are created for the gut and thus often fail to achieve growth on SNM3. Here, 3 of the 5 models fail to achieve growth.</p>
<p>As a result, we cannot analyze them. However, it must be noted that when analyzing a community, this may be less relevant, as other members can provide missing metabolites through interactions. On the other hand, it may be well known that this species can grow on the medium hence this indicates some problems with the metabolic model.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">standardized_models</span> <span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">models</span><span class="p">:</span>
    <span class="n">standardized_models</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">set_default_configs_and_snm3_medium</span><span class="p">(</span><span class="n">model</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">standardized_models</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;Growth </span><span class="si">{model.id}</span><span class="s2">: {model.slim_optimize()}&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Growth CarveMe_SNM_gapfilled_model: 0.0
Growth himodel: 0.0
Growth Staphylococcus_epidermidis_ATCC_12228: 0.5746259500792691
Growth DP_83VPs_KB5: 0.0
Growth iYS854: 3.044356975357284
</pre></div></div>
</div>
<p>In order to allow analysis and to fix errors, we implement several “gap filling” strategies. This adds metabolites to the medium or reactions to the model so that it can grow.</p>
</div>
<div class="section" id="Gapfilling">
<h2>Gapfilling<a class="headerlink" href="#Gapfilling" title="Permalink to this headline">¶</a></h2>
<p>To achieve growth on a general predefined medium, we deploy two strategies: * <strong>Gapfill medium</strong>: Since other members of the community could provide different metabolites that are currently unmodeled, the medium is less strict from the point of view of the community. Hence we may extend the medium by the minimal number of metabolites such that the organism can grow. * <strong>Gapfill model:</strong> We can add reactions to the model such that it can grow on the given medium i.e.&nbsp;reactions such that the
model can synthesize the missing metabolites.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">standardized_models</span><span class="p">:</span>
    <span class="n">old_medium</span> <span class="o">=</span> <span class="n">model</span><span class="o">.</span><span class="n">medium</span>
    <span class="n">m2</span><span class="p">,</span> <span class="n">extension</span> <span class="o">=</span> <span class="n">gapfill_medium</span><span class="p">(</span><span class="n">model</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;We need to add following metabolites to the medium: </span><span class="si">{extension}</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;Growth </span><span class="si">{m2.id}</span><span class="s2">: {m2.slim_optimize()}&quot;</span><span class="p">)</span>
    <span class="n">m2</span><span class="o">.</span><span class="n">medium</span> <span class="o">=</span> <span class="n">old_medium</span> <span class="c1"># Rest for next step</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
We need to add following metabolites to the medium: [&#39;EX_fe3dcit_e&#39;]
Growth CarveMe_SNM_gapfilled_model: 1.20618744198712
We need to add following metabolites to the medium: [&#39;EX_3ump_e&#39;]
Growth himodel: 1.706848711893427
We need to add following metabolites to the medium: []
Growth Staphylococcus_epidermidis_ATCC_12228: 0.5746259500792691
We need to add following metabolites to the medium: [&#39;EX_LalaDgluMdap_e&#39;, &#39;EX_ile__L_e&#39;, &#39;EX_metox_e&#39;]
Growth DP_83VPs_KB5: 0.28275198236758425
We need to add following metabolites to the medium: []
Growth iYS854: 3.044356975357284
</pre></div></div>
</div>
<p>Nice, this worked. It also provides us with information about missing metabolites. Note that the solution to this problem is not unique e.g.&nbsp;for the first result an equivalent result would be obtained by adding EX_fe3_e.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">standardized_models</span><span class="p">:</span>
    <span class="n">m2</span><span class="p">,</span> <span class="n">extension</span> <span class="o">=</span> <span class="n">gapfill_model</span><span class="p">(</span><span class="n">model</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;We need to add following reactions to the model: {[e for e in extension]}&quot;</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;Growth </span><span class="si">{m2.id}</span><span class="s2">: {m2.slim_optimize()}&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
c:\users\manug\desktop\ncmw projects\ncmw\ncmw\setup_models\setup.py:66: UserWarning: The model still has no growth... . We try an alternative
  warn(&#34;The model still has no growth... . We try an alternative&#34;)
c:\users\manug\desktop\ncmw projects\ncmw\ncmw\setup_models\setup.py:81: UserWarning: The model still has no growth... . We greadily add sink reactions...
  warn(&#34;The model still has no growth... . We greadily add sink reactions...&#34;)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
We need to add following reactions to the model: [&#39;SK_fe3dcit_e&#39;]
Growth CarveMe_SNM_gapfilled_model: 1.206187441987119
We need to add following reactions to the model: [&lt;Reaction DINSt2 at 0x1c7b7b52220&gt;]
Growth himodel: 1.2333198465405384
We need to add following reactions to the model: []
Growth Staphylococcus_epidermidis_ATCC_12228: 0.5746259500792709
We need to add following reactions to the model: [&lt;Reaction ACHBS at 0x1c7b1192d00&gt;, &lt;Reaction ASPK at 0x1c7b35fd370&gt;, &lt;Reaction DAPE at 0x1c7ba1746a0&gt;, &lt;Reaction DHAD2 at 0x1c7ba174a00&gt;, &lt;Reaction DHDPRy at 0x1c7ba174a30&gt;, &lt;Reaction DHDPS at 0x1c7ba174a60&gt;, &lt;Reaction HSDx at 0x1c7b2009790&gt;, &lt;Reaction KARA2 at 0x1c7b2009dc0&gt;, &lt;Reaction ASAD_1 at 0x1c7b9c7cd00&gt;]
Growth DP_83VPs_KB5: 0.27989919666526347
We need to add following reactions to the model: []
Growth iYS854: 3.044356975357276
</pre></div></div>
</div>
<p>Nice, this worked too! Notice that the results are quite similar but slightly different. Just adding reactions to a model is, of course, questionable i.e.&nbsp;these reactions should have some empirical evidence. One can look e.g.&nbsp;in KEGG if genes for certain reactions are present for a given species.</p>
</div>
<div class="section" id="Cleaning-the-model">
<h2>Cleaning the model<a class="headerlink" href="#Cleaning-the-model" title="Permalink to this headline">¶</a></h2>
<p>There are typically several blocked reactions or mass charge violations within a metabolic model. Thus before checking the quality we may want to remove these e.g.&nbsp;using the FastCC algorithm.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">consistent_models</span> <span class="o">=</span><span class="p">[]</span>
<span class="n">reports</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">standardized_models</span><span class="p">:</span>
    <span class="n">consistent_model</span><span class="p">,</span> <span class="n">report</span> <span class="o">=</span> <span class="n">create_consistent_model</span><span class="p">(</span><span class="n">model</span><span class="p">)</span>
    <span class="n">consistent_models</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">consistent_model</span><span class="p">)</span>
    <span class="n">reports</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">report</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
C:\Users\manug\.conda\envs\ncmw\lib\site-packages\cobra\core\group.py:107: UserWarning: need to pass in a list
  warn(&#34;need to pass in a list&#34;)
C:\Users\manug\.conda\envs\ncmw\lib\site-packages\cobra\core\group.py:107: UserWarning: need to pass in a list
  warn(&#34;need to pass in a list&#34;)
</pre></div></div>
</div>
<p>Lets look at one of the report returned.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">reports</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Blocked reactions</th>
      <th>Metabolite formula problems</th>
      <th>Mass charge balance violations</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Original Model</th>
      <td>246</td>
      <td>0</td>
      <td>122</td>
    </tr>
    <tr>
      <th>Consistent model</th>
      <td>0</td>
      <td>0</td>
      <td>117</td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
<p>As expected we can see some improvements.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="k">for</span> <span class="n">model</span> <span class="ow">in</span> <span class="n">consistent_models</span><span class="p">:</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">f</span><span class="s2">&quot;Growth </span><span class="si">{model.id}</span><span class="s2">: {model.slim_optimize()}&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Growth CarveMe_SNM_gapfilled_model: 0.0
Growth himodel: 0.0
Growth Staphylococcus_epidermidis_ATCC_12228: 0.5746259500792683
Growth DP_83VPs_KB5: 0.0
Growth iYS854: 3.0443569753572817
</pre></div></div>
</div>
<p>And that’s it. Now we may first check the model’s quality i.e.&nbsp;using <strong>memote</strong>. Next, we can start analyzing and comparing results for the models, or continue to create a community model from these. This also serves as a small overview of what the “ncmw_setup” can do.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="analysis.html" class="btn btn-neutral float-right" title="Analysis" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="readme.html" class="btn btn-neutral float-left" title="Overview" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2022, manuelgloeckler

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>